File: Blanquita-Frontend/src/models/BobinaServilleta/BobinaServilleta.js
export const UnidadBobinaServilletaItem = (data) => ({
  CodigoBobina: data.CodigoBobina,
  IdFormatoSubBobina: data.IdFormatoSubBobina,
  PesoBrutoKg: data.PesoBrutoKg ?? null,
  GramajeGr: data.GramajeGr ?? null
});

export const BobinaServilletaItem = (data) => ({
  Unidades: (data.Unidades ?? []).map(UnidadBobinaServilletaItem)
});

export const IngresoBobinaServilletaRequest = (idProveedor, idTipoBobinaServilleta, bobinas) => ({
  IdProveedor: idProveedor,
  IdTipoBobinaServilleta: idTipoBobinaServilleta,
  Bobinas: (bobinas ?? []).map(BobinaServilletaItem)
});

export const IngresoBobinaServilletaResponse = (data) => ({
  FechaRecepcion: data.FechaRecepcion,
  CantidadBobinasServilleta: data.CantidadBobinasServilleta,
  CantidadUnidades: data.CantidadUnidades
});

File: Blanquita-Backend/app/Models/BobinaServilleta/BobinaServilleta.py

from datetime import date

from pydantic import BaseModel


class UnidadBobinaServilletaItem(BaseModel):
    CodigoBobina: str
    IdFormatoSubBobina: int
    PesoBrutoKg: float | None = None
    GramajeGr: float | None = None


class BobinaServilletaItem(BaseModel):
    Unidades: list[UnidadBobinaServilletaItem]


class IngresoBobinaServilletaRequest(BaseModel):
    IdProveedor: int
    IdTipoBobinaServilleta: int
    Bobinas: list[BobinaServilletaItem]


class IngresoBobinaServilletaResponse(BaseModel):
    FechaRecepcion: date
    CantidadBobinasServilleta: int
    CantidadUnidades: int


class TipoBobinaServilletaIngreso(BaseModel):
    IdTipoBobinaServilleta: int
    NombreTipoBobinaServilleta: str